Extensions/retags.js

```diff
@@ -25,17 +25,29 @@ XKit.extensions.retags = {
 
 	observer: new MutationObserver(function(ms){
 		ms.forEach(function(m){
-			XKit.extensions.retags.tag($(m.addedNodes).filter(XKit.extensions.retags.selectors));
+			if ($(m.addedNodes).find('.note').length > 0) {
+				XKit.extensions.retags.tag($(m.addedNodes).find('.note').filter(XKit.extensions.retags.selectors));
+			} else {
+				XKit.extensions.retags.tag($(m.addedNodes).filter(XKit.extensions.retags.selectors));
+			}
 		});
 	}),
 
 	add_toggle: function(){
 		var toggle = 'retags_toggle_'+this.blog_name;
-		this.html_toggle.appendTo('.ui_notes_switcher .part-toggle');
+		if (XKit.browser().mobile) {
+			this.html_toggle.appendTo('.primary-nav');
+			XKit.tools.add_css('label.retags .binary_switch_label {font-size:15px; color:white; padding-bottom:15px; }','retags_mobile_label');
+		} else {
+			this.html_toggle.appendTo('.ui_notes_switcher .part-toggle');
+		}
 		$('#retags-toggle').change(function(){
 			if ($(this).prop('checked')) {
 				localStorage.setItem(toggle,'true');
 				XKit.extensions.retags.css_toggle.appendTo('head');
+				if (XKit.browser().mobile) {
+					XKit.extensions.retags.mobile_toggle.appendTo('head');
```

Comment:
Keeping everything uniform. css_toggle does the same thing, and this is just the mobile addition to it.